controllers/authorino_controller.go

```diff
@@ -443,73 +478,96 @@ func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino
 	}
 
 	// get ServiceAccount from authorino instance namespace
-	sa := &k8score.ServiceAccount{}
-	saName := resourcePrefixName + "-authorino"
-	saNsdName := namespacedName(authorino.Namespace, saName)
-	if err := r.Get(context.TODO(), saNsdName, sa); err != nil {
-		if errors.IsNotFound(err) {
-			// ServiceAccount doesn't exit - create one
-			ctrl.SetControllerReference(authorino, sa, r.Scheme)
-			sa.Name = saName
-			sa.Namespace = authorino.Namespace
-			if err := r.Client.Create(context.TODO(), sa); err != nil {
-				return r.wrapErrorWithStatusUpdate(
-					logger, authorino, r.setStatusFailed(api.AuthorinoUnableToCreateServiceAccount),
-					fmt.Errorf("failed to create %s ServiceAccount, err: %v", saName, err),
-				)
-			}
+	if sa, err := r.createAuthorinoServiceAccount(authorino); err != nil {
+		return err
+	} else {
+		// creates the ClusterRoleBinding/RoleBinding depending on type of installation
+		if err := r.reconcileAuthorinoRoleBinding(authorinoManagerClusterRoleBindingNameSuffix, authorinoClusterRoleName, *sa, authorino.Spec.ClusterWide, authorino); err != nil {
+			return err
 		}
-		return r.wrapErrorWithStatusUpdate(
-			logger, authorino, r.setStatusFailed(api.AuthorinoUnableToGetServiceAccount),
-			fmt.Errorf("failed to get %s ServiceAccount, err: %v", saName, err),
-		)
+		// creates the K8s Auth ClusterRoleBinding (for Authorino's Kubernetes TokenReview and SubjectAccessReview features)
+		// Disclaimer: this has nothing to do with kube-rbac-proxy, but to authn/authz features of Authorino that also require cluster scope role bindings
+		if err := r.reconcileAuthorinoRoleBinding(authorinoK8sAuthClusterRoleBindingNameSuffix, authorinoK8sAuthClusterRoleName, *sa, true, authorino); err != nil {
+			return err
+		}
+
+		// creates leader election role
+		// TODO: describe what this leader election is
+		return r.leaderElectionPermission(authorino, *sa)
 	}
+}
 
-	// creates the ClusterRoleBinding/RoleBinding depending on type of installation
-	// if ClusterWide true - ClusterRoleBinding
-	// if ClusterWide false - RoleBinding
-	var binding client.Object = authorinoResources.GetAuthorinoClusterRoleBinding(authorinoClusterRoleName, sa.Name, sa.Namespace)
-	if !authorino.Spec.ClusterWide {
-		binding = authorinoResources.GetAuthorinoRoleBinding(authorinoClusterRoleName, sa.Name, sa.Namespace)
-		binding.SetNamespace(authorino.Namespace)
+func authorinoRoleBindingName(crName, roleBindingNameSuffix string, clusterScoped bool) string {
```

Comment:
Could we move this to the resource package under the [k8s_rbac.go](https://github.com/Kuadrant/authorino-operator/blob/main/pkg/resources/k8s_rbac.go) file?